lualib/resty/kong/peer_conn.lua

```diff
@@ -5,36 +5,41 @@ local get_request      = base.get_request
 local errmsg           = base.get_errmsg_ptr()
 local C                = ffi.C
 local ffi_str          = ffi.string
-local get_phase        = ngx.get_phase
+local subsystem        = ngx.config.subsystem
 local NGX_ERROR        = ngx.ERROR
 
 local error            = error
 
 
-ffi.cdef[[
-int
-ngx_http_lua_kong_ffi_get_last_peer_connection_cached(ngx_http_request_t *r,
-    char **err);
-]]
-
-
-local function get_last_peer_connection_cached()
-    if get_phase() ~= "balancer" then
-        error("get_last_peer_connection_cached() can only be called in balancer phase")
-    end
-
-    local r = get_request()
-    if not r then
-        error("no request found")
+local get_last_peer_connection_cached
+
+
+if subsystem == "http" then
+    require "resty.core.phase"  -- for ngx.get_phase
+    ffi.cdef[[
+    int ngx_http_lua_kong_ffi_get_last_peer_connection_cached(ngx_http_request_t *r,
+        char **err);
+    ]]
+    
+    get_last_peer_connection_cached = function()
+        local ngx_phase = ngx.get_phase
```

Comment:
We should move this line below `require "resty.core.phase"`

Reply:
done